webapp/src/ts/services/training-cards.service.ts

```diff
@@ -0,0 +1,127 @@
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+
+import { XmlFormsService } from '@mm-services/xml-forms.service';
+import { TrainingCardsComponent } from '@mm-modals/training-cards/training-cards.component';
+import { DbService } from '@mm-services/db.service';
+import { GlobalActions } from '@mm-actions/global';
+import { ModalService } from '@mm-modals/mm-modal/mm-modal';
+import { SessionService } from '@mm-services/session.service';
+import { RouteSnapshotService } from '@mm-services/route-snapshot.service';
+
+export const TRAINING_PREFIX: string = 'training';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TrainingCardsService {
+  private globalActions;
+
+  constructor(
+    private store: Store,
+    private xmlFormsService: XmlFormsService,
+    private dbService: DbService,
+    private modalService: ModalService,
+    private sessionService: SessionService,
+    private routeSnapshotService: RouteSnapshotService,
+  ) {
+    this.globalActions = new GlobalActions(store);
+  }
+
+  private getAvailableTrainingForms(xForms, userCtx) {
+    const today = new Date();
+
+    return xForms
+      .map(xForm => ({
+        id: xForm._id,
+        code: xForm.internalId,
+        startDate: xForm.start_date,
```

Comment:
@jkuester I've been thinking... you know how app forms has a property.json file that looks like this:
```
{
    "icon": "icon-pregnancy",
    "title": "Pregnancy registration",
    "context": {
        "person": true,
        "place": false,
        "expression": "contact.type === 'person'"
    }
}
```

We can do the same for training cards to set: `start_date`, `duration` and `user_roles`, when using cht-conf, what do you think? If yes, then how about if we put these in the first level:
```
{
    "icon": "icon-pregnancy",
    "title": "Pregnancy registration",
    "start_date": 12345663221,
    "duration": 60,
    "user_roles": [ "nurse", "supervisor" ]
    "context": {
        "person": true,
        "place": false,
        "expression": "contact.type === 'person'"
    }
}
```
Or is it better inside the `context`? Like this: 
```
{
    "icon": "icon-pregnancy",
    "title": "Pregnancy registration",
    "context": {
        "person": true,
        "place": false,
        "expression": ".........",
        "start_date": 12345663221,
        "duration": 60,
        "user_roles": [ "nurse", "supervisor" ]
    }
}
```
In the last example, then we'll access the properties here like: `xForm.context.start_date`

Reply:
I've decided to put these properties inside the context, work better with cht-conf:
```
{
    "icon": "icon-pregnancy",
    "title": "Pregnancy registration",
    "context": {
        "person": true,
        "place": false,
        "expression": ".........",
        "start_date": 12345663221,
        "duration": 60,
        "user_roles": [ "nurse", "supervisor" ]
    }
}
```